server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -478,8 +478,14 @@ private void init()
           jp = objectMapper.getFactory().createParser(future.get());
           final JsonToken nextToken = jp.nextToken();
           if (nextToken == JsonToken.START_OBJECT) {
-            QueryInterruptedException e = jp.getCodec().readValue(jp, QueryInterruptedException.class);
-            throw e;
+            QueryInterruptedException cause = jp.getCodec().readValue(jp, QueryInterruptedException.class);
+            //case we get an exception with an unknown message.
+            if (cause.isNotKnown()) {
+              throw new QueryInterruptedException(QueryInterruptedException.UNKNOWN_EXCEPTION, cause.getMessage(), host);
+            } else {
+              throw  new QueryInterruptedException(cause, host);
```

Comment:
this will be a QueryInterruptedException in a QueryInterruptedException.

I think that's ok, but is it intended?